R/generate_code.R

```diff
@@ -0,0 +1,90 @@
+#' Reverse engineer code for creation of a [`dm`]
+#'
+#' `cdm_paste` takes an existing `dm` and produces the code necessary for its creation
+#'
+#' @inheritParams cdm_add_pk
+#' @param select Boolean, default `FALSE`. If `TRUE` will try to produce code for reducing to necessary columns.
+#' @param env The environment to search for the tables for the creation of the `dm`. Defaults to the global environment.
+#'
+#' @details At the very least (if no keys exist in the given [`dm`]) a `dm()` statement is produced that -- when executed --
+#' produces the same `dm`. In addition, the code for setting the existing primary keys as well as the relations between the
+#' tables is produced. Should the tables not be available in the given environment, a warning is issued. The code won't work
+#' as is in this case. If `select = TRUE`, the column names of the `dm` tables are compared with the column names
+#' of the tables of the same name in the given environment. For each table:
+#' 1. If the `dm` tables have less columns and all of them exist in the tables in the given environment, appropriate `cdm_select()` statements are created
+#' 1. If the `dm` tables have extra columns, the table is skipped and a warning is issued.
+#'
+#' @return Code for producing the given `dm`
+#'
+#' @export
+cdm_paste <- function(dm, select = FALSE, env = .GlobalEnv) {
+  check_dm(dm)
+  check_no_filter(dm)
+  check_not_zoomed(dm)
+
+  table_names <- src_tbls(dm)
+  pks <- cdm_get_all_pks(dm)
+  fks <- cdm_get_all_fks(dm)
+
+  tables_exist <- set_names(map_lgl(table_names, ~ exists(..1, envir = env)), table_names)
+  # code for including the tables
+  code <- paste0("dm(", paste(table_names, collapse = ", "), ") %>%\n")
```

Comment:
```suggestion
  code <- paste0("dm(", paste(tick_if_needed(table_names), collapse = ", "), ") %>%\n")
```

Borrow from tibble if necessary.